jupyter/d2l-java/chapter_linear-networks/linear-regression.ipynb

```diff
@@ -0,0 +1,871 @@
+    "and thus there is only one (global) minimum.\n",
+    "However, for more complicated models, like deep networks,\n",
+    "the loss surfaces contain many minima.\n",
+    "Fortunately, for reasons that are not yet fully understood,\n",
+    "deep learning practitioners seldom struggle to find parameters\n",
+    "that minimize the loss *on training data*.\n",
+    "The more formidable task is to find parameters\n",
+    "that will achieve low loss on data\n",
+    "that we have not seen before,\n",
+    "a challenge called *generalization*.\n",
+    "We return to these topics throughout the book.\n",
+    "\n",
+    "\n",
+    "\n",
+    "### Making Predictions with the Learned Model\n",
+    "\n",
+    "\n",
+    "Given the learned linear regression model\n",
+    "$\\hat{\\mathbf{w}}^\\top \\mathbf{x} + \\hat{b}$,\n",
+    "we can now estimate the price of a new house\n",
+    "(not contained in the training data)\n",
+    "given its area $x_1$ and age (year) $x_2$.\n",
+    "Estimating targets given features is\n",
+    "commonly called *prediction* and *inference*.\n",
+    "\n",
+    "We will try to stick with *prediction* because\n",
+    "calling this step *inference*,\n",
+    "despite emerging as standard jargon in deep learning,\n",
+    "is somewhat of a misnomer.\n",
+    "In statistics, *inference* more often denotes\n",
+    "estimating parameters based on a dataset.\n",
+    "This misuse of terminology is a common source of confusion\n",
+    "when deep learning practitioners talk to statisticians.\n",
+    "\n",
+    "\n",
+    "### Vectorization for Speed\n",
+    "\n",
+    "When training our models, we typically want to process\n",
+    "whole minibatches of examples simultaneously.\n",
+    "Doing this efficiently requires that we vectorize the calculations\n",
+    "and leverage fast linear algebra libraries\n",
+    "rather than writing costly for-loops in Java.\n",
+    "\n",
+    "To illustrate why this matters so much,\n",
+    "we can consider two methods for adding vectors.\n",
+    "To start we instantiate two $10000$-dimensional vectors\n",
+    "containing all ones.\n",
+    "In one method we will loop over the vectors with a Java `for` loop.\n",
+    "In the other method we will rely on a single call to `djl`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%maven ai.djl:api:0.5.0\n",
+    "%maven org.slf4j:slf4j-api:1.7.26\n",
+    "%maven org.slf4j:slf4j-simple:1.7.26\n",
+    "\n",
+    "// See https://github.com/awslabs/djl/blob/master/mxnet/mxnet-engine/README.md\n",
+    "// for more MXNet library selection options\n",
+    "%maven ai.djl.mxnet:mxnet-engine:0.5.0\n",
+    "%maven ai.djl.mxnet:mxnet-native-auto:1.7.0-a"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%loadFromPOM\n",
```

Comment:
use:
%maven tech.tablesaw:tablesaw-jsplot:0.30.4

Reply:
I tried doing that earlier, but I have to additionally import com.google.common.collect.ImmutableList and I get a java.langNoClassDefFoundError for tech.tablesaw.columns.numbers.DoubleColumnType.

Reply:
I'll leave it as the loadFromPOM for now